### PR TITLE
[SHELL32] Give opportunity to rename file/folder on its creation

### PR DESCRIPTION
**Pull request by zecarlos1957**
Revert the hack from commit ed4b16b7f (PR #4950).
Real fix to this issue.

The CNewMenu::SelectNewItem method starts with a request to notify the View object about a new item by calling SHChangeNotify. After creating and obtaining the new PIDL by calling ILFindLastID, the CDefView::SelectItem method is called. This method fails right from the start when calling LV_FindItemByPidl which returns -1, thus preventing the user from being given the opportunity to change the name displayed by default.

This failure is due to the fact that this object has not yet been added to the CDefView::m_ListView. For some reason the call to SHChangeNotify did not call CDefView::OnChangeNotify with eventID SHCNE_CREATE.

Thanks to @yagoulas  by the big help.

### CODE REVIEW

**Comment by HBelusca** [comment]
@zecarlos1957 Do you know which commit(s) introduced that "hack" ?

**Comment by zecarlos1957** [comment]
@HBelusca  Is ed4b16b7ff3

File: dll/win32/shell32/CNewMenu.cpp [hunk 1]
```diff
@@ -467,7 +467,7 @@ HRESULT CNewMenu::CreateNewFolder(LPCMINVOKECOMMANDINFO lpici)
         return E_FAIL;
 
     /* Show and select the new item in the def view */
-    SelectNewItem(SHCNE_MKDIR, SHCNF_PATHW, wszName, TRUE);
+    SelectNewItem(SHCNE_MKDIR, SHCNF_PATHW | SHCNF_FLUSH, wszName, TRUE);
```
[the remark this answers is not in the document]
Comment (zecarlos1957) [reply dll/win32/shell32/CNewMenu.cpp]: Done. Thanks

File: dll/win32/shell32/CNewMenu.cpp [hunk 2]
```diff
@@ -418,7 +418,7 @@ HRESULT CNewMenu::SelectNewItem(LONG wEventId, UINT uFlags, LPWSTR pszName, BOOL
         dwSelectFlags |= SVSI_EDIT;
 
     /* Notify the view object about the new item */
-    SHChangeNotify(wEventId, uFlags, (LPCVOID) pszName, NULL);
+    SHChangeNotify(wEventId, uFlags | SHCNF_FLUSH, (LPCVOID) pszName, NULL);
```
Comment (HBelusca) [inline dll/win32/shell32/CNewMenu.cpp:421]:
```suggestion
    SHChangeNotify(wEventId, uFlags | SHCNF_FLUSH, (LPCVOID)pszName, NULL);
```

**Review by HBelusca — APPROVED** [review]

**Review by yagoulas — APPROVED** [review]

**Pull request merged**